typing/subst.ml

```diff
@@ -278,6 +278,12 @@ let rec typexp copy_scope s ty =
       | Tpackage(p, fl) ->
           Tpackage(modtype_path s p,
                     List.map (fun (n, ty) -> (n, typexp copy_scope s ty)) fl)
+       | Tfunctor(lbl, us, (p, fl), ty) ->
+          let us' = Ident.Unscoped.refresh us in
+          let fl = List.map (fun (n, ty) -> (n, typexp copy_scope s ty)) fl in
```

Comment:
(Harping on my previous comment, this could be `let tpackage' = tpackage copy_scope p fl in ...`, shared wit hthe case above.)